src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -418,41 +450,63 @@ class IpexCommunicationService extends AgentService {
     ).map((key) => previousExnGrantMsg.exn.e.acdc.e?.[key]?.s);
     allSchemaSaids.push(schemaSaid);
 
-    const op = await Agent.agent.multiSigs.multisigAdmit(
+    const { op } = await this.multisigService.multisigAdmit(
       holder.signifyName,
       previousExnGrantMsg.exn.d as string,
       allSchemaSaids,
       multisigExn
     );
 
     const schema = await this.props.signifyClient.schemas().get(schemaSaid);
-    await this.saveAcdcMetadataRecord(
-      previousExnGrantMsg.exn.e.acdc.d,
-      previousExnGrantMsg.exn.e.acdc.a.dt,
-      schema.title,
-      connectionId,
-      schemaSaid
-    );
+    const credentialPending =
+      await this.credentialStorage.getCredentialMetadata(
+        previousExnGrantMsg.exn.e.acdc.d
+      );
 
-    this.props.eventService.emit<AcdcStateChangedEvent>({
-      type: AcdcEventTypes.AcdcStateChanged,
-      payload: {
-        credentialId,
-        status: CredentialStatus.PENDING,
-      },
-    });
+    if (!credentialPending) {
+      await this.saveAcdcMetadataRecord(
+        previousExnGrantMsg.exn.e.acdc.d,
+        previousExnGrantMsg.exn.e.acdc.a.dt,
+        schema.title,
+        connectionId,
+        schemaSaid
+      );
 
-    const pendingOperation = await this.operationPendingStorage.save({
-      id: op.name,
-      recordType: OperationPendingRecordType.ExchangeReceiveCredential,
+      this.props.eventService.emit<AcdcStateChangedEvent>({
+        type: AcdcEventTypes.AcdcStateChanged,
+        payload: {
+          credentialId,
+          status: CredentialStatus.PENDING,
+        },
+      });
+
+      const pendingOperation = await this.operationPendingStorage.save({
+        id: op.name,
+        recordType: OperationPendingRecordType.ExchangeReceiveCredential,
+      });
+      this.signifyNotificationService.addPendingOperationToQueue(
+        pendingOperation
+      );
+    }
+
+    const notifications = await this.notificationStorage.findAllByQuery({
+      exnSaid: exn?.exn.e.exn.p,
     });
-    Agent.agent.signifyNotifications.addPendingOperationToQueue(
-      pendingOperation
-    );
-    Agent.agent.signifyNotifications.deleteNotificationRecordById(
-      id,
-      notifRecord.a.r as NotificationRoute
-    );
+
+    if (notifications.length) {
+      const notificationRecord = notifications[0];
+      if (
+        notificationRecord.linkedGroupRequests &&
+        Object.keys(notificationRecord.linkedGroupRequests).length
```

Comment:
I think we can remove the if statement - not only will this already be set (or we wouldn't be calling this function), but the default value is `{}`